### PR TITLE
Added ability to pass-in parameters into Resource constructors, fixes #429

### LINKED ISSUE #429: Allow parameters to be passed to the resource class constructor on add_resource()
As a similar example, one can do this in flask with [Pluggable Views](http://flask.pocoo.org/docs/0.10/views/)
```
class RenderTemplateView(View):
    def __init__(self, template_name):
        self.template_name = template_name
    def dispatch_request(self):
        return render_template(self.template_name)

app.add_url_rule('/about', view_func=RenderTemplateView.as_view(
    'about_page', template_name='about.html'))
```

In my current use case of flask-restful, I need my resource to get a parameter ( storing data that will be requested later via REST ) and AFAIK it s not currently possible with add_resource ( or I cannot find a way... )

add_resource uses as_view internally, so I am guessing it is probably doable...

### PR DESCRIPTION
**Pull request by mikehwang**

### CODE REVIEW

**Comment by tommyjcarpenter** [comment]
I vouch for this PR!

**Comment by svenstaro** [comment]
Could you add a paragraph or two to the docs about this to make users aware of why they should use this?

**Comment by mikehwang** [comment]
Added to docs/intermediate-usage.rst

**Comment by svenstaro** [comment]
LGTM

File: docs/intermediate-usage.rst [hunk 1]
```diff
@@ -244,3 +244,33 @@ Finally, we used the :class:`fields.Url` field type. ::
...
 urls of the objects in the ``links`` sub-object.  Passing ``absolute=True``
 ensures that the generated urls will have the hostname included.
+
+
+Passing Constructor Parameters Into Resources
+---------------------------------------------
+Your :class:`Resource` implementation may require outside dependencies. Those
+dependencies are best passed-in through the constructor to loosely couple each
+other. The :meth:`Api.add_resource` method has two keyword arguments:
+`resource_class_args` and `resource_class_kwargs`. Their values will be forwarded
```
Comment (joshfriend) [inline docs/intermediate-usage.rst]: needs to be wrapped with double backticks to get formatted as `monospaced`

**Comment by mikehwang** [comment]
Done :+1:

**Comment by joshfriend** [comment]
:+1:

**Comment by mikehwang** [comment]
Awesome!! When will this new version get pushed to pypi?

**Comment by tommyjcarpenter** [comment]
🍻

**Comment by joshfriend** [comment]
@mikehwang @tommyjcarpenter: #448

**Pull request merged**